pal03377/Athena-New
# Fix download programming repositories

### Motivation and Context



@bassner had some issues running the script and I'm addressing those issues in this PR.

### Description


- Use `axios` for requests now (this probably also lowers the node version requirement to node 16)
- Improve general error handling/logging
- Remove parallelization in favor of a more long running script that is more transparent with what is going on
- Add participation ids to the evaluation data 
- Download the repositories in batches of 25 instead of having one request downloading potentially ~1000 repos at a time 
- Write `download-<timestamp>.txt` and `submission-<timestamp>.txt` for logging downloaded/failed exercise/participation ids

### Steps for Testing


1. `cd playground`
2. `npm install`
3. Use the data from below in `data/evaluation/programming-exercises.json`
4. Run `npm run export:artemis:3-download-programming-repositories`
     a. Login with instructor test account on Artemis (production)
     b. Select all exercises
     c. Download
5. It should now run in batches

```
{
  "TEST": {
    "semester": "WS20/21",
    "course_id": 86,
    "exercises": [
      {
        "id": 10573,
        "title": "TEST 01",
        "participations": [
          7945271,
          8082848,
          7936574
        ],
        "is_exam_exercise": 0
      },
      {
        "id": 10573,
        "title": "TEST 02",
        "participations": [
          7945271,
          8082848,
          7936574
        ],
        "is_exam_exercise": 0
      }
    ]
  }
}
```

### Screenshots

![image_720](https://github.com/ls1intum/Athena/assets/5898705/6920a634-62ef-4313-a702-ba2855adfede)
